website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,826 @@
+    });
+
+    self.serialized = ko.pureComputed(function () {
+        // TODO(lyndsysimon): Test this.
+        var self = this;
+        var data = {};
+
+        $.each(self.pages(), function (_, page) {
+            $.each(page.questions(), function (_, question) {
+                data[question.id] = {value: question.value()};
+            });
+        });
+
+        return {
+            schema_name: self.draft().metaSchema.name,
+            schema_version: self.draft().metaSchema.version,
+            schema_data: data
+        };
+    }.bind(self));
+
+    self.isDirty = ko.pureComputed(function() {
+        // TODO(lyndsysimon): Test this.
+        var request = self.lastSaveRequest();
+        if (request === undefined) {
+            // Note that if a save request has not happened, the form is considered dirty.
+            return true;
+        }
+        if (request.payload === undefined) {
+            // If the payload is not attached, the .done() handler hasn't been called;
+            //      therefore the request has not successfully completed.
+            return false;
+        }
+        return JSON.stringify(request.payload) !== JSON.stringify(self.serialized());
+    });
+
+    /* The last autosave request that was sent to the server. */
+    self.lastSaveRequest = ko.observable();
+    self.lastSaveAge = ko.pureComputed(function () {
+        // TODO(lyndsysimon): Test this.
+        var request = self.lastSaveRequest();
+        if (request === undefined) {
+            // not yet saved
+            return;
+        } else if (request.completedDate === undefined) {
+            // request in flight
+            return -1;
+        } else {
+            return (new Date() - request.completedDate);
+        }
+    });
+
+
+    self.lastSaved = ko.computed(function() {
+        var request = self.lastSaveRequest();
+        if (request !== undefined) {
+            if (request.completedDate !== undefined) {
+                return request.completedDate.toGMTString();
+            } else {
+                return 'pending';
+            }
+        } else {
+            return 'never';
+        }
+    });
+
+    self.canRegister = ko.computed(function() {
+        var draft = self.draft();
+        return draft && draft.isApproved;
+    });
+
+    self.iterObject = $osf.iterObject;
+
+    // TODO: better extensions system?
+    self.extensions = {
+        'osf-upload': editorExtensions.Uploader
+    };
+};
+/**
+ * Load draft data into the editor
+ *
+ * @param {Draft} draft
+ **/
+RegistrationEditor.prototype.init = function(draft) {
+    var self = this;
+
+    self.draft(draft);
+    var metaSchema = draft.metaSchema;
+
+    var schemaData = {};
+    if (draft) {
+        schemaData = draft.schemaData || {};
+    }
+
+    // Set currentPage to the first page
+    self.currentPage(self.draft().pages()[0]);
+
+    self.serialized.subscribe(self.autoSave.bind(self));
```

Comment:
A few notes here:

1. This seems to be the mechanism for calling autosave when the data changes, correct?
2. Instead of bind, you can use a second argument to `subscribe`.  Per http://knockoutjs.com/documentation/observables.html

> The subscribe function accepts three parameters: callback is the function that is called whenever the notification happens, target (optional) defines the value of this in the callback function, and event (optional; default is "change") is the name of the event to receive notification for.